src/transformers/models/t5/modeling_t5.py

```diff
@@ -632,7 +632,7 @@ def forward(
                 )
 
             self_attn_past_key_value = past_key_value[:2]
-            cross_attn_past_key_value = past_key_value[2:]
+            cross_attn_past_key_value = past_key_value[-2:]
```

Comment:
Why change that?

Reply:
This is for using `past_key_values` in the encoder. 

Because if we decide to use past_key_values in the encoder, the `past_key_values[2:]` will cause IndexError. Since encoder only needs past key and value to calculate self-attention,  the length of encoder past_key_values should be 2.Therefore I change it to `[-2:]` to avoid this error in the encoder and not affect the decoder.

But I think we can discuss this in a new issue later. Let's fix the decoder first.